python-bindings/pythonbindings.cpp

```diff
@@ -26,6 +26,14 @@ namespace py = pybind11;
 using namespace bls;
 using std::vector;
 
+class PythonGIL {
```

Comment:
pybind11 already has a class that (I believe) does this: `pybind11::gil_scoped_release`

Reply:
This release type is considerably faster than gil_scoped_release for situations where you won't be reacquiring the GIL. Using gil_scoped_release negates most of the benefit of this optimization because the runtimes of the calls are small in an absolute sense. We just make a large number of times.